cmd/modern/main.go

```diff
@@ -45,10 +45,25 @@ func main() {
 		cmdparser.Initialize(initializeCallback)
 		rootCmd.Execute()
 	} else {
+		initializeEnvVars()
 		legacyCmd.Execute(version)
 	}
 }
 
+// initializeEnvVars intializes SQLCMDSERVER, SQLCMDUSER and SQLCMDPASSWORD
+// if the currentContext is set. This env variables do not have any effect
+// if user provides this info via backward compatibility syntax
+func initializeEnvVars() {
+	initializeCallback()
+	if config.CurrentContextName() != "" {
+		server, username, password := config.GetCurrentContextInfo()
+		os.Setenv("SQLCMDSERVER", server)
```

Comment:
Good point.  In case user already has env variables set and also has current-context set, then shouldn't the current-context info take precedence here?  If current-context doesn't have user info, then I agree that existing env variables (SQLCMDUSER, SQLCMDPASSWORD) should be left untouched.

Reply:
I don't feel great about this new behavior taking precedence, for existing scripts etc.  If a user has set this env vars, I think we should honor that.  And only set them here if they are not already set.

Reply:
environment variables should always take precedence over config files, and command line switches take precedence over environment variables.